python/pyarrow/serialization.py

```diff
@@ -30,27 +31,83 @@
 except ImportError:
     cloudpickle = builtin_pickle
 
+if six.PY2:
+    JSONDecodeError = ValueError
+else:
+    JSONDecodeError = json.decoder.JSONDecodeError
 
 # ----------------------------------------------------------------------
 # Set up serialization for numpy with dtype object (primitive types are
 # handled efficiently with Arrow's Tensor facilities, see
 # python_to_arrow.cc)
 
+
+class DTypeJsonEncoder(json.JSONEncoder):
+    @classmethod
+    def object_converter(cls, data):
+        if isinstance(data, tuple):
+            return {
+                '__tuple__': True,
+                'data': tuple(cls.object_converter(e) for e in data)
+            }
+        elif isinstance(data, np.dtype):
+            return {
+                '__dtype__': True,
+                'data': data.str
+            }
+        elif isinstance(data, list):
+            return [
+                cls.object_converter(e)
+                for e in data
+            ]
+        elif isinstance(data, dict):
+            return {
+                key: cls.object_converter(value)
+                for key, value in data.items()
+            }
+        else:
+            return data
+
+    def encode(self, obj):
+        return super(DTypeJsonEncoder, self).encode(self.object_converter(obj))
+
+
+def _dtype_json_object_hook(obj):
+    if '__tuple__' in obj:
+        return tuple(obj['data'])
+    elif '__dtype__' in obj:
+        return np.dtype(obj['data'])
+    else:
+        return obj
+
+
 def _serialize_numpy_array_list(obj):
     if obj.dtype.str != '|O':
         # Make the array c_contiguous if necessary so that we can call change
         # the view.
         if not obj.flags.c_contiguous:
             obj = np.ascontiguousarray(obj)
-        return obj.view('uint8'), obj.dtype.str
+
+        if obj.dtype.fields is not None:
+            dtype = DTypeJsonEncoder().encode(dict(obj.dtype.fields))
```

Comment:
I think it's overkill to bring json into this and customize the encoder. You could simply use pickle, which Numpy dtypes support natively (you could also use `dtype.descr`, but it won't work with all dtypes).